homeassistant/components/schlage/lock.py

```diff
@@ -0,0 +1,82 @@
+"""Platform for Schlage lock integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from pyschlage.lock import Lock
+
+from homeassistant.components.lock import LockEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, MANUFACTURER
+from .coordinator import SchlageDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Schlage WiFi locks based on a config entry."""
+    coordinator: SchlageDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities(
+        SchlageLockEntity(coordinator=coordinator, device_id=device_id)
+        for device_id in coordinator.data.locks
+    )
+
+
+class SchlageLockEntity(CoordinatorEntity[SchlageDataUpdateCoordinator], LockEntity):
+    """Schlage lock entity."""
+
+    def __init__(
```

Comment:
https://developers.home-assistant.io/docs/core/entity?_highlight=has_entity_name#has_entity_name-true-mandatory-for-new-integrations

Please see above

has_entity_name True (Mandatory for new integrations)